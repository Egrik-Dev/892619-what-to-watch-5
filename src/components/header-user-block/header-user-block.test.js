import React from 'react';
import renderer from 'react-test-renderer';
import HeaderUserBlock, {HeaderUserBlock as HeaderUserBlockWithoutStore} from './header-user-block';
import {Route, Router} from 'react-router-dom';
import browserHistory from '../../browser-history';
import {Provider} from "react-redux";
import configureStore from "redux-mock-store";

it(`Should HeaderUserBlockWithoutStore render correctly with Authorization Status`, () => {
  const tree = renderer
    .create(
        <Router history={browserHistory}>
          <Route>
            <HeaderUserBlockWithoutStore
              authorizationStatus={`AUTH`}
              avatar={`https://assets.htmlacademy.ru/intensives/javascript-3/avatar/3.jpg`}
            />
          </Route>
        </Router>
    ).toJSON();

  expect(tree).toMatchSnapshot();
});

it(`Should HeaderUserBlockWithoutStore render correctly without Authorization Status`, () => {
  const tree = renderer
    .create(
        <Router history={browserHistory}>
          <Route>
            <HeaderUserBlockWithoutStore
              authorizationStatus={`NO_AUTH`}
              avatar={``}
            />
          </Route>
        </Router>
    ).toJSON();

  expect(tree).toMatchSnapshot();
});

describe(`Render connected to store component`, () => {
  const mockStore = configureStore([]);
  let store = null;
  let HeaderUserBlockComponent = null;

  beforeEach(() => {
    store = mockStore({
      USER: {
        authorizationStatus: `AUTH`,
        avatar: `https://assets.htmlacademy.ru/intensives/javascript-3/avatar/3.jpg`
      }
    });

    HeaderUserBlockComponent = renderer.create(
        <Provider store={store}>
          <Router history={browserHistory}>
            <Route>
              <HeaderUserBlock />
            </Route>
          </Router>
        </Provider>
    );
  });

  it(`Should HeaderUserBlock connected to store render correctly`, () => {
    expect(HeaderUserBlockComponent.toJSON()).toMatchSnapshot();
  });
});
