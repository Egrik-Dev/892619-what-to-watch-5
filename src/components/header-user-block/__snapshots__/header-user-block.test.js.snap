// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render connected to store component Should HeaderUserBlock connected to store render correctly 1`] = `
<div
  className="user-block"
>
  <a
    href="/mylist"
    onClick={[Function]}
  >
    <div
      className="user-block__avatar"
    >
      <img
        alt="User avatar"
        height="63"
        src="https://assets.htmlacademy.ru/intensives/javascript-3/avatar/3.jpg"
        width="63"
      />
    </div>
  </a>
</div>
`;

exports[`Should HeaderUserBlockWithoutStore render correctly with Authorization Status 1`] = `
<div
  className="user-block"
>
  <a
    href="/mylist"
    onClick={[Function]}
  >
    <div
      className="user-block__avatar"
    >
      <img
        alt="User avatar"
        height="63"
        src="https://assets.htmlacademy.ru/intensives/javascript-3/avatar/3.jpg"
        width="63"
      />
    </div>
  </a>
</div>
`;

exports[`Should HeaderUserBlockWithoutStore render correctly without Authorization Status 1`] = `
<div
  className="user-block"
>
  <a
    className="user-block__link"
    href="/login"
    onClick={[Function]}
  >
    Sign in
  </a>
</div>
`;
